### PR TITLE
Run unit tests in pull requests

### PR DESCRIPTION
**Pull request by MoritzSalein**
Changed the GitHub Action for executing QTAF on chrome and renamed the Action.

### CODE REVIEW

File: .github/workflows/TestCompilation.yml
Comment (csvtuda) [inline .github/workflows/TestCompilation.yml:1]: While the changes are much appreciated, I think this action should undergo a total rework:
1. It does not make sense to run `mvn clean test` on push to both `main` and `develop`. The `main` branch is used only for backing up the latest release code, which never actually gets there without going through the tests run on release: https://github.com/Qytera-Gmbh/QTAF/blob/0b40bfb043b7ae17a56e69917afe697e34d2ed5d/.github/workflows/PublishMavenCentral.yml#L36-L51 This job should probably be moved into its own action that runs tests when trying to merge a pull request into `develop`.
3. I would much prefer a matrix approach for building the docker images to reduce code duplication. Something like this:
```yml
  build_java_image:
    runs-on: ubuntu-latest
    timeout-minutes: ${{ env.TIMEOUT_MINUTES }}
    strategy:
      matrix:
        version: [ 17, 18, 19 ]
        include:
          - version: 17
            command: sh ./scripts/docker/build-java17-image.sh
          - version: 18
            command: sh ./scripts/docker/build-java18-image.sh
          - version: 19
            command: sh ./scripts/docker/build-java19-image.sh
    steps:
      - uses: actions/checkout@v2
      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v2
      - name: Cache Docker layers
        uses: actions/cache@v2
        with:
          path: /tmp/.buildx-cache
          key: ${{ runner.os }}-buildx-${{ github.sha }}
          restore-keys: |
            ${{ runner.os }}-buildx-
      - name: Build docker image
        run: ${{ matrix.command }}
```

**Review by csvtuda — CHANGES_REQUESTED** [review]

**Comment by MoritzSalein** [comment]
@csvtuda Do you think the new version of the Action could work?

File: .github/workflows/TestCompilation.yml [hunk 1]
```diff
@@ -1,59 +1,45 @@
...
+            port: 4445
+    services:
+      chrome:
+        image: selenium/standalone-chrome
+        ports:
+          - 4444:4444
+      firefox:
+        image: selenium/standalone-firefox
+        ports:
+          - 4445:4444
```
Comment (csvtuda) [inline .github/workflows/TestCompilation.yml:30]: Nit: I think we could also do 
```yml
ports:
  - ${{ matrix.port }}:4444
```
here (not quite sure), but it's fine as is.

**Comment by csvtuda** [comment]
Looks good to me! I think we should switch to the reusable unit test action as well for the testing part, though.
- Unit test action: https://github.com/Qytera-Gmbh/QTAF/blob/eecf699f67a37cfec22f059db5da4fad8ab5ab02/.github/workflows/Test.yml
- Example call: https://github.com/Qytera-Gmbh/QTAF/blob/eecf699f67a37cfec22f059db5da4fad8ab5ab02/.github/workflows/Publish.yml#L13-L17

I think providing a remote driver doesn't really affect QTAF's unit tests, I've never had to provide it. So it should probably be fine to switch to the test action.

**Comment by MoritzSalein** [comment]
I tried to insert the reusable test Action in my Action. Also I renamed it, to clarify that it only runs on the develop branch.

**Review by csvtuda — APPROVED** [review]

**Pull request merged**